Podcast/SeriesDetailHeaderView.swift

```diff
@@ -64,145 +62,219 @@ class SeriesDetailHeaderView: UIView {
+            make.trailing.equalToSuperview().inset(17.5)
+            make.height.equalTo(70.0)
+        }
+        
+        viewSeparator.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.height.equalTo(1.0)
+            make.width.equalTo(tagsView.snp.width)
+            make.bottom.equalTo(tagsView.snp.top)
+        }
         
-        tagsView.addSubview(relatedTagsLabel)
-    
-        addSubview(infoView)
-        addSubview(tagsView)
-        addSubview(viewSeparator)
     }
     
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
     
     func setSeries(series: Series) {
-        let titleX = 2 * padding + imageHeight
         titleLabel.text = series.title
-        titleLabel.frame = CGRect(x: titleX, y: padding, width: frame.width - titleX - padding, height: 0)
         UILabel.adjustHeightToFit(label: titleLabel, numberOfLines: 3)
         publisherLabel.text = series.author
-        publisherLabel.frame = CGRect(x: titleX, y: titleLabel.frame.maxY + marginPadding, width: frame.width - titleX - padding, height: 0)
         UILabel.adjustHeightToFit(label: publisherLabel, numberOfLines: 1)
         
-        let lastEpisodeLabelY = publisherLabel.frame.maxY > imageView.frame.maxY ? publisherLabel.frame.maxY + padding : imageView.frame.maxY + padding
-        lastEpisodeLabel.frame = CGRect(x: padding, y: lastEpisodeLabelY, width: 0, height: 0)
-        lastEpisodeLabel.sizeToFit()
-        
-        lastEpisodeDateLabel.text = Date.formatDateDifferenceByLargestComponent(fromDate: series.lastUpdated, toDate: Date())
-        lastEpisodeDateLabel.frame = CGRect(x: padding, y: lastEpisodeLabel.frame.maxY + marginPadding, width: frame.width - 2 * padding, height: 0)
-        UILabel.adjustHeightToFit(label: lastEpisodeDateLabel, numberOfLines: 1)
-        
-        subscribeButton.frame = CGRect(x: padding, y: lastEpisodeDateLabel.frame.maxY + padding, width: subscribeWidth, height: subscribeHeight)
-        settingsButton.frame = CGRect(x: 2 * padding + subscribeWidth, y: 0, width: smallButtonSideLength, height: smallButtonSideLength)
+        // Share button not in current design
         shareButton.frame = CGRect(x: frame.width - padding - smallButtonSideLength, y: 0, width: smallButtonSideLength, height: smallButtonSideLength)
         
         shareButton.center.y = subscribeButton.center.y
-        settingsButton.center.y = subscribeButton.center.y
-        relatedTagsLabel.frame.origin.y = viewSeparator.frame.maxY + padding / 2
         
-        infoView.frame = CGRect(x: 0, y: 0, width: frame.width, height: subscribeButton.frame.maxY + padding)
-        viewSeparator.frame = CGRect(x: 0, y: infoView.frame.maxY, width: frame.width, height: separatorHeight)
+        subscribeButtonChangeState(isSelected: series.isSubscribed)
+
+        if let url = series.largeArtworkImageURL{
```

Comment:
you don't need a if let here... you can just do imageView.setImageAsynchronouslyWithDefaultImage(url: url), backgroundImageView.setImageAsynchronouslyWithDefaultImage(url: url) because .setImageAsynchronouslyWithDefaultImage takes in an optional url